### PR TITLE
Remove CopyOnWriteArrayList usage

### PR DESCRIPTION
**Pull request by RoiEXLab**
Follow up of #1939
This change is compatible. I tested savegames from the current stable and tested a networked game

### CODE REVIEW

File: src/main/java/games/strategy/engine/data/GameData.java [hunk 1]
```diff
@@ -74,9 +74,9 @@
   private String gameName;
   private Version gameVersion;
   private int diceSides;
-  private transient List<TerritoryListener> territoryListeners = new CopyOnWriteArrayList<>();
-  private transient List<GameDataChangeListener> dataChangeListeners = new CopyOnWriteArrayList<>();
-  private transient List<GameMapListener> gameMapListeners = new CopyOnWriteArrayList<>();
+  private transient List<TerritoryListener> territoryListeners = new ArrayList<>();
+  private transient List<GameDataChangeListener> dataChangeListeners = new ArrayList<>();
+  private transient List<GameMapListener> gameMapListeners = new ArrayList<>();
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/data/GameData.java]: Can't `GameData` be accessed by multiple threads concurrently?  I don't know if listeners are ever actually added/removed/iterated concurrently, but these fields are not protected by `readWriteLock`, so there's now the potential for `ConcurrentModificationException` to be thrown while iterating.

Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/data/GameData.java]: Good point. I don't think I went over this one closely enough. Any non-UI classes have the potential to be accessed by multiple threads. We may need to revert this PR.

**Comment by ron-murhammer** [comment]
@RoiEXLab @ssoloff I think we should probably revert this PR for now. Thoughts?

**Comment by RoiEXLab** [comment]
I believe proposing a PR which reverts the changes in non-UI classes would be a better idea.
We can then check if individual cases are actually accessed from various threads.

**Comment by ssoloff** [comment]
@ron-murhammer I agree with @RoiEXLab.  I think this change is small enough that just the non-UI classes can be reverted in a separate PR, and then they can be scrutinized later.  If @RoiEXLab codes up the PR, I can review it and compare it to this one just to make sure we didn't miss any of the non-UI cases.

**Pull request merged**